packages/@aws-cdk/aws-lambda-python/test/integ.function.requirements.removed.ts

```diff
@@ -1,69 +0,0 @@
-/// !cdk-integ pragma:ignore-assets
-import * as fs from 'fs';
-import * as os from 'os';
-import * as path from 'path';
-import { Runtime } from '@aws-cdk/aws-lambda';
-import { App, CfnOutput, Stack, StackProps } from '@aws-cdk/core';
-import { Construct, ConstructOrder } from 'constructs';
-import * as lambda from '../lib';
-
-/*
- * Stack verification steps:
- * aws lambda invoke --function-name <function name> --invocation-type Event --payload $(base64 <<<'"OK"') response.json
- */
-
-class TestStack extends Stack {
-  public readonly dependenciesAssetHash: string;
-
-  constructor(scope: Construct, id: string, props?: StackProps) {
-    super(scope, id, props);
-
-    const fn = new lambda.PythonFunction(this, 'function', {
-      entry: workDir,
-      runtime: Runtime.PYTHON_3_9,
-    });
-
-    new CfnOutput(this, 'Function', {
-      value: fn.functionName,
-    });
-
-    // Find the asset hash of the dependencies
-    this.dependenciesAssetHash = (fn.node.findAll(ConstructOrder.POSTORDER)
-      .find(c => c.node.path.endsWith('Code')) as any)
-      .assetHash;
-  }
-}
-
-// This is a special integration test that synths twice to show that docker
-// picks up a change in requirements.txt
-
-// Create a working directory for messing around with requirements.txt
-const workDir = fs.mkdtempSync(path.join(os.tmpdir(), 'cdk-integ'));
-fs.copyFileSync(path.join(__dirname, 'lambda-handler', 'index.py'), path.join(workDir, 'index.py'));
-const requirementsTxtPath = path.join(workDir, 'requirements.txt');
-
-// Write a requirements.txt with an extraneous dependency (colorama)
-const beforeDeps = 'certifi==2020.6.20\nchardet==3.0.4\nidna==2.10\nurllib3==1.26.7\nrequests==2.26.0\nPillow==8.4.0\ncolorama==0.4.3\n';
-fs.writeFileSync(requirementsTxtPath, beforeDeps);
-
-// Synth the first time
-const app = new App();
-const stack1 = new TestStack(app, 'cdk-integ-lambda-python-requirements-removed1');
-app.synth();
-
-// Then, write a requirements.txt without the extraneous dependency and synth again
-const afterDeps = 'certifi==2020.6.20\nchardet==3.0.4\nidna==2.10\nurllib3==1.26.7\nrequests==2.26.0\nPillow==8.4.0\n';
-fs.writeFileSync(requirementsTxtPath, afterDeps);
```

Comment:
This test kept failing for me, regardless of how I modified the `afterDeps` value. In the traceback, I could only see one of the commands being run, which makes me think it has something to do with how synth + caching occurs with the bundling process.

In the previous version, dependency installation was occurring during the Docker build process (compared to Docker image run now), so maybe that has something to do with it? Not sure.

Reply:
Noticed this in the logs:

```
### `cdk synthesize`
aws-cdk: Synthesizes the CDK app and produces a cloud assembly to a designated output (defaults to `cdk.out`)
aws-cdk: Typically you don't interact directly with cloud assemblies. They are files that include everything
aws-cdk: needed to deploy your app to a cloud environment. For example, it includes an AWS CloudFormation
aws-cdk: template for each stack in your app, and a copy of any file assets or Docker images that you reference
aws-cdk: in your app.
```

which to me reads like: `app.synth()` in the above referenced stack wouldn't actually create the asset; that would be during `cdk deploy`.